### ISSUE TITLE
Connect: ListRootClusters RPC does not return LoggedInUser

### ISSUE THREAD

**Issue by ravicious** [gravitational/teleport#20670]
We used to always return it when calling `newAPIRootCluster`:

https://github.com/gravitational/teleport/blob/e6d2485d664303aae9735bba5dda5fc1fd821530/lib/teleterm/apiserver/handler/handler_clusters.go#L87-L88

Whereas now it's available only after `EnrichWithDetails` has been called:

https://github.com/gravitational/teleport/blob/927f52164fc27b49f0f18b21ee45e9c81389b726/lib/teleterm/apiserver/handler/handler_clusters.go#L87-L94

https://github.com/gravitational/teleport/blob/927f52164fc27b49f0f18b21ee45e9c81389b726/lib/teleterm/clusters/cluster.go#L122

This has a couple of repercussions, as for example the UI now briefly behaves as if the user was not logged in. Only after `GetCluster` is received, the logged in user is added and the UI opens the workspace.

https://user-images.githubusercontent.com/27113/214549340-aad3aeda-3b30-454e-8abb-de65a33c4dc7.mov

I feel like fixing this might also fix #20603.

**Comment by ravicious** [comment]
@gzdunek Oh, another idea: in integration/proxy/teleterm_test.go we can keep the tests which actually test the db access (`"root cluster"` and `"leaf cluster"`). Then in integration/teleterm_test.go we'd put the other tests.

The setup for them would no longer need to create a leaf cluster. However, I just realized that at the moment the helpers do not support creating _just_ the root cluster. Leave that to me since I already implemented this once, for now integration/teleterm_test.go can just copy the same setup.

**Comment by gzdunek** [comment]
Good idea! So I guess we can get rid of creating a new user in `TestGetCluster` as we have our own cluster (?)

**Comment by ravicious** [comment]
It'd be best to create it anyway to make the test independent of other tests (and save someone a headache and refactoring effort in the future).
